pkg/vm/engine/disttae/txn_table.go

```diff
@@ -570,11 +570,26 @@ func (tbl *txnTable) Ranges(ctx context.Context, exprs []*plan.Expr) (ranges [][
 	if len(tbl.blockInfos) == 0 {
 		return
 	}
+
+	// for dynamic parameter, sustitute param ref and const fold cast expression here to improve performance
+	// temporary solution, will fix it in the future
+	newExprs := make([]*plan.Expr, len(exprs))
+	bat := batch.NewWithSize(0)
+	bat.Zs = []int64{1}
```

Comment:
you can use  batch.EmptyForConstFoldBatch to instead of batch.New....

Reply:
fixed